lib/util/enumerate-property-names.js

```diff
@@ -7,11 +7,18 @@
 const { CALL, CONSTRUCT, READ } = require("@eslint-community/eslint-utils")
 const unprefixNodeColon = require("./unprefix-node-colon")
 
+/**
+ * @typedef {Object} DeprecatedInfo
+ * @property {string} since
+ * @property {string|{ name: string, supported: string }[]|null} replacedBy
+ */
+/** @typedef {import('@eslint-community/eslint-utils').TraceMap<DeprecatedInfo>} DeprecatedInfoTraceMap */
+
 /**
  * Enumerate property names of a given object recursively.
- * @param {import('../unsupported-features/types.js').SupportVersionTree} trackMap The map for APIs to enumerate.
+ * @param {import('../unsupported-features/types.js').SupportVersionTree | DeprecatedInfoTraceMap} trackMap The map for APIs to enumerate.
  * @param {string[]} [path] The path to the current map.
- * @param {WeakSet} [recursionSet] A WeakSet used to block recursion (eg Module, Module.Module, Module.Module.Module)
+ * @param {WeakSet<any>} [recursionSet] A WeakSet used to block recursion (eg Module, Module.Module, Module.Module.Module)
```

Comment:
```suggestion
 * @param {WeakSet<unknown>} [recursionSet] A WeakSet used to block recursion (eg Module, Module.Module, Module.Module.Module)
```

Would this work?

Reply:
Unfortunatly not, WeakSet requires the property to extend `object` :disappointed:

I can, however change the type to expect one of the trace maps!
![image](https://github.com/eslint-community/eslint-plugin-n/assets/2230835/0073127f-ac3a-42a1-a57f-c67f989d1b31)